sources/commands/Disable.ts

```diff
@@ -70,6 +70,13 @@ export class DisableCommand extends Command<Context> {
   async removePosixLink(installDirectory: string, binName: string) {
     const file = path.join(installDirectory, binName);
     try {
+      const currentTarget = await fs.promises.realpath(file);
+
+      if (binName.includes(`yarn`) && currentTarget.match(/[/\\]switch[/\\]bin[/\\]/)) {
```

Comment:
We should run the `realpath` only for Yarn
```suggestion
      if (binName.includes(`yarn`) && /[/\\]switch[/\\]bin[/\\]/.test(await fs.promises.realpath(file))) {
```